### DISCUSSION TITLE
Valetudo 2023.04.0

### DISCUSSION THREAD

**Discussion by Hypfer** [Hypfer/Valetudo#1805] [Release Discussions]
<div align="center">
    <img src="https://raw.githubusercontent.com/Hypfer/Valetudo/master/assets/logo/valetudo_logo_with_name.svg" width="800" alt="valetudo">
    <p align="center"><h2>2023.04.0</h2></p>
</div>

The Timer Update + Roborock S7 Pro Ultra support

## Timers

Inbuilt timers have been greatly improved with the latest release.

### Pre-Actions
Since we now have supported robots where the Water setting actually makes a noticeable difference, timers needed a way to control those settings. For that, Pre-Actions have been introduced.

![image](https://user-images.githubusercontent.com/974410/233849469-87656478-8ff7-4f23-8f7d-cbcc8d83a90d.png)

If you enable any of those, they will execute the command to e.g. set the fan, water or operation mode before starting the configured action for the timer.

For example, you can now have:
- one timer at 10am that cleans all rooms with tiled floors with a high water setting

followed by
- another timer at 11.30am that cleans all wooden floors with less water.

### Custom Labels

Timers can now be named with a custom label to make it easy to keep track of which one does what

![image](https://user-images.githubusercontent.com/974410/233849485-49f171df-9f17-4cb3-a074-3ff5211d5988.png)

### Manual execution

It is now possible to execute a timer manually by pressing the play button.

![image](https://user-images.githubusercontent.com/974410/233849500-97660c1c-7eb7-49d9-98ca-36a8093b35c4.png)

As the timer doesn't have to be enabled to be manually executable, this can also be (ab-)used to recall specific cleaning scenarios without having to select the same segments and settings over and over again.

![image](https://user-images.githubusercontent.com/974410/233849507-33610505-d7ac-46b7-9b9c-9a90d1709802.png)

### UTC conversion bugfix

Lastly, a bug was fixed that caused the timers to be executed on the wrong day if UTC conversion crossed the midnight line.

## S7 Pro Ultra support

Thanks to your donations, I was able to buy a Roborock S7 Pro Ultra.

![image](https://user-images.githubusercontent.com/974410/233849519-144e9873-b778-4975-8d87-69dac2b04e2b.png)

The robot is now supported by Valetudo. It can be rooted and used right away.

## Misc

1. MQTT and NTP editor now automatically extract the host from pasted URLs for more convenience
2. Headlines in the sidebar menu no longer overlap
2. CarpetModeControl is now also available for the W10

## The usual

If you want to see Valetudo on more robots and/or like this release, you might want to consider donating:

[https://github.com/sponsors/Hypfer](https://github.com/sponsors/Hypfer)

[https://builder.dontvacuum.me/donations.txt](https://builder.dontvacuum.me/donations.txt)

### Features

- **timers**: Introduce custom labels for timers [`eadff02`](http://github.com/Hypfer/Valetudo/commit/eadff02564f36de7749363a09d32d1ff071cb3ea)
- **timers**: Allow manual execution of stored timers [`a5f920e`](http://github.com/Hypfer/Valetudo/commit/a5f920e01c2b5cbf47b17cbabf073fe151b97235)
- **timers**: Introduce pre_actions [`6092d75`](http://github.com/Hypfer/Valetudo/commit/6092d75a5145f256c52920c5c880829cb518d330)
- **vendor.roborock**: Ultra dock consumables [`1f624db`](http://github.com/Hypfer/Valetudo/commit/1f624db25f54d0ab8549595f95130855180d67b7)
- **vendor.roborock**: Add support for the S7 Pro Ultra [`87636ba`](http://github.com/Hypfer/Valetudo/commit/87636ba921eb65b4bffde2125dc70cc75ee19b70)
- **ui**: Improve UX when configuring MQTT or NTP hosts [`cf982c0`](http://github.com/Hypfer/Valetudo/commit/cf982c0e626d4996227d0e291207f90c8ce73528)

### Fixes

- **ui**: Fix Timer DoW shifting when encountering the midnight line during UTC conversion [`a922877`](http://github.com/Hypfer/Valetudo/commit/a9228771ac59bd46f4ca0e406915760590aa4297)
- **vendor.dreame**: Add DreameCarpetModeControlCapability to DreameMopValetudoRobots such as the W10 [`c39cd7e`](http://github.com/Hypfer/Valetudo/commit/c39cd7e2a676ff378390f39eeb9abc9fac3a78f8)
- **vendor.dreame**: Fix detergent consumable reset [`8636e0c`](http://github.com/Hypfer/Valetudo/commit/8636e0ca201c99e3d55d1e4863425148803ee826)
- **vendor.viomi**: Fix parser freezing when encountering unexpected segment name data [`fc73347`](http://github.com/Hypfer/Valetudo/commit/fc73347177709d7b0ff3ad02c3b2e7ae3c024c49)
- **ui**: Hide clear button if there is no GoTo target [`7c6c0a9`](http://github.com/Hypfer/Valetudo/commit/7c6c0a959c327c9558a2cd1ed078e45747ca572f)
- **ui**: Fix sidebar menu subheader overlapping [`c394a54`](http://github.com/Hypfer/Valetudo/commit/c394a54358f78726b63cb24833acf359414b1b1d)

### Refactoring

- Don't override inbuilt identifiers [`3eb9548`](http://github.com/Hypfer/Valetudo/commit/3eb9548c5c630c255fb950eafa2130a590ccdbd2)

### Chores

- **release**: 2023.04.0 [`9b0668c`](http://github.com/Hypfer/Valetudo/commit/9b0668cc7ffbac2157023113a11b94eefd85d51c)
- **ui**: MQTT Settings cleanup [`7327236`](http://github.com/Hypfer/Valetudo/commit/7327236b411ba9e30de73bfe80de7f2425cc6631)
- Bump engine to node v18 [`de8935c`](http://github.com/Hypfer/Valetudo/commit/de8935ce51ce62330be0d0116bef0759578add87)
- **vendor.dreame**: Wording [`321ff10`](http://github.com/Hypfer/Valetudo/commit/321ff10b73143ea0895352fc249ddc7818c5a84b)

<hr /><em>This discussion was created from the release <a href='https://github.com/Hypfer/Valetudo/releases/tag/2023.04.0'>Valetudo 2023.04.0</a>.</em>

**Comment by SA0TAY** [comment]
Heh. I was just about to open a bug report since the .upx files were missing, breaking links on the tutorials, when I saw the timestamp. Waited 30 more seconds, and voilà.
